### ISSUE TITLE
Compatibility with TypeScript 3.5.2

### ISSUE THREAD

**Issue by rysavyjan** [mathjax/MathJax-src#309]
I'm not able to compile latest source code version from GitHub with TypeScript 3.5.2, I'm getting errors like

```
mathjax3-ts/output/svg/Wrappers/TextNode.ts:37:84 - error TS2562: Base class expressions cannot reference class type parameters.
mathjax3-ts/util/Options.ts:227:17 - error TS2538: Type 'symbol' cannot be used as an index type.
...
```

What TS version are you using?

**Comment by dpvc** [comment]
Sorry, I thought I had responded to this last week, but apparently I didn't press the "comment" button.

The `package-lock.json` file has typescript at 2.3.4, but your first error message doesn't appear until version 2.5.  I'm not sure when the second error message first shows up.

The "base class" issue is one that I knew about, and have plans to address, but wasn't planning to do so until after 3.0.0 is released.  It is a type tracking issue, not a code generation issue.  The code being produced by typescript version 2.3.4 is correct, it is just that typescript is having trouble knowing what types things are (though it doesn't realize that until version 2.6).

I have just now pushed a branch (`tsc3`) that compiles with typescript 3.5.2.  It works around the issues, but a better solution for the mixins is needed.  I have a technique that I think will work, but I don't want to hold up version 3.0.0 for it, since it is already very delayed as it is.
